lib/rules/no-reservered-keys.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @fileoverview Prevent overwrite reserved keys
+ * @author Armano
+ */
+'use strict'
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+const RESERVER_NAMES = new Set(require('../utils/vue-reserved.json'))
+const SCOPE_NAMES = new Set(['props', 'computed', 'data', 'methods'])
+
+function create (context) {
+  const reservedNames = RESERVER_NAMES
+  const scopeNames = SCOPE_NAMES
+
+  const options = context.options[0] || {}
+  if (options.reserved) {
+    options.reserved.forEach(name => {
+      reservedNames.add(name)
+    })
+  }
+  if (options.scope) {
+    options.scope.forEach(name => {
+      scopeNames.add(name)
+    })
+  }
+
+  // ----------------------------------------------------------------------
+  // Public
+  // ----------------------------------------------------------------------
+
+  return utils.executeOnVueComponent(context, (obj) => {
+    utils.iterateProperties(obj, scopeNames, (name, node, groupName) => {
+      if (groupName === 'data' && name.substr(0, 1) === '_') {
+        context.report({
+          node: node,
+          message: "Field '{{name}}' which start with _ in data is reserved.",
```

Comment:
What do you think about this message: `Keys starting with with "_" are reserved in "data" group.` ?